File: test10/equi_filter.py
#!/usr/bin/python3

import sys
import re

for line in sys.stdin:
    lll=[]
    words=re.split('\s+',line)
    for word in words:
        ll={}
        for ch in word:
            ch=ch.lower()
            if ch in ll:
                ll[ch]+=1
            else:
                ll[ch]=1
        

        l=set(ll.values())
        if len(l)==1:
            lll.append(word)
    newline=" ".join(lll)
    print(newline)

File: test10/equi_filter.pl
#!/usr/bin/perl 

use List::MoreUtils qw(uniq);


@lines=<STDIN>;
foreach my $line(@lines){
	@ll=split " ", $line;
	@newll=();
	for (my $i=0;$i<@ll;$i++){
		@lll=split "",$ll[$i];
		%words=();
		foreach my $char(@lll){
			$char=lc($char);
			$words{$char}+=1;
		}
		@countt= values %words;
 		@unique= uniq @countt;
		$size=@unique;
		if ($size == 1 ){
			push(@newll,$ll[$i]);
		}
	}
	$newline=join(" ",@newll);
	print "$newline\n";
}
